datanode/dehistory/store/store_ipfs.go

```diff
@@ -0,0 +1,303 @@
+package store
```

Comment:
the file was getting unwieldy so I split it into 2, with ipfs related stuff in the seconds, however, I think theres been a fair bit of refactrogin since then and the split is no longer adhered to and the files are short so I've compressed back into the one file store.go

Reply:
fair enough, not a big deal